### PR TITLE
Fix SCTP support on FreeBSD & NetBSD

### PR DESCRIPTION
**Pull request by jailbird777**
Unlike Linux, Solaris, and Illumos (and probably others), the 2 BSDs that still support SCTP didn't put SCTP into its own library; they put it into libc.

They, unlike Linux, don't set `SOL_SCTP` for us. The official method appears to be calling `getprotobyname("sctp")` & `endprotoent()`, with `getprotobyname()` returning a struct that has a `p_proto` entry. This all reads from `/etc/protocols`. However, SCTP is assigned 132 by IANA, so it's 132 everywhere, so I just set `SOL_SCTP` to 132 if it's not already set.

### CODE REVIEW

**Comment by jailbird777** [comment]
LIBSCTP_LD didn't seem to be actually used at the top-level, so I removed it there.

If you don't like the SOL_SCTP hack, I can change the #ifndef to actually do the getprotoname() lookup.

File: librb/src/commio.c [hunk 1]
```diff
@@ -357,6 +357,9 @@ rb_setsockopt_reuseaddr(rb_fde_t *F)
 }
 
 #ifdef HAVE_LIBSCTP
+#ifndef SOL_SCTP
+#define SOL_SCTP 132
+#endif
 static int
```
Comment (aaronmdjones) [inline librb/src/commio.c:363]: Please insert blank lines before and after this ifndef block for enhanced readability.

**Review by aaronmdjones — CHANGES_REQUESTED** [review]

File: librb/src/commio.c (hunk 1 above)
Comment (jailbird777) [reply librb/src/commio.c:365]: Done, thanks!

**Review by aaronmdjones — APPROVED** [review]

**Comment by aaronmdjones** [comment]
(Context: from discussion on IRC; various system headers (and lksctp-tools) define `IPPROTO_SCTP` to the same value)

**Review by aaronmdjones — APPROVED** [review]

**Review by jillest — APPROVED** [review]
Looks good (untested)

**Pull request merged**